### ISSUE TITLE
`await` in non-async function should report as syntax error.

### ISSUE THREAD

**Issue by gaeulbyul** [denoland/deno_lint#1139]
### Lint Name

SyntaxError?

### Code Snippet

```ts
function invalid() {
  await Promise.resolve(true)
}

invalid()
```

### Expected Result
It should report as syntax error

### Actual Result
Deno doesn't find any problem with that code.

### Version

* deno 1.31.2 (release, x86_64-pc-windows-msvc)
* v8 11.0.226.19
* typescript 4.9.4

**Comment by magurotuna** [comment]
You can get a wanted error from TypeScript by running `deno check your_script.ts`.

```
$ cat script.ts
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ File: script.ts
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ function invalid() {
   2   │   await Promise.resolve(true)
   3   │ }
   4   │
   5   │ invalid()
───────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

$ deno check script.ts
Check file:///private/tmp/script.ts
error: TS1308 [ERROR]: 'await' expressions are only allowed within async functions and at the top levels of modules.
  await Promise.resolve(true)
  ~~~~~
    at file:///private/tmp/script.ts:2:3

TS1356 [ERROR]:     Did you mean to mark this function as 'async'?
    function invalid() {
             ~~~~~~~
        at file:///private/tmp/script.ts:1:10
```

Basically deno_lint attempts to discover possibly problematic pieces of code that TypeScript is not covering. So putting `await `in non-async functions is what TypeScript is responsible for, not deno_lint.

**Comment by dsherret** [comment]
This will also fail at runtime:

```
> deno run main.js
error: Uncaught SyntaxError: Unexpected reserved word
  await Promise.resolve(true)
  ^
    at <anonymous> (file:///V:/deno_lint/main.js:2:3)
```
